pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/MinionTaskUtils.java

```diff
@@ -206,4 +183,66 @@ public static boolean extractMinionAllowDownloadFromServer(TableConfig tableConf
     }
     return defaultValue;
   }
+
+  /**
+   * Returns the validDocID bitmap from the server whose local segment crc matches both crc of ZK metadata and
+   * deepstore copy.
+   */
+  @Nullable
+  public static RoaringBitmap getValidDocIdFromServerMatchingCrc(String tableNameWithType, String segmentName,
+      String validDocIdsType, MinionContext minionContext, String originalSegmentCrcFromTaskGenerator,
+      String crcFromDeepStorageSegment) {
```

Comment:
hmm.. should task return early `if (!originalSegmentCrcFromTaskGenerator.equals(crcFromDeepStorageSegment)` 

when these two matched, then we can call this util method to get a validDocIds bitmap from servers that can match the expected crc.

Reply:
Hmm yeah good point! Though we should fix scenarios where SegmentZKMetadata and deepstore-copy-CRC are not same as a separate issue.